src/components/surety-guard.tsx

```diff
@@ -21,7 +22,7 @@ export const SuretyGuard: FC<{
     return (
         <Popover width={300} trapFocus withArrow shadow="md" opened={opened} onChange={close} closeOnClickOutside>
             <Popover.Target>
-                <ActionIcon size="sm" variant="subtle" color="red" onClick={open} disabled={opened}>
+                <ActionIcon size="sm" variant="subtle" color="red" onClick={open} disabled={opened || blockAction}>
```

Comment:
the `blockAction` prop seems superfluous to me.  Instead of relying on the SuretyGuard to not call the functions we provided, we can either not render it, or render it but have our function be a no-op. 

Addendum: as I was typing this, I realized that I guess there is a case for wanting to render the button in a disabled state.  Maybe we rename the property to be `disabled`?